### PR TITLE
feat: run script builder

### PR DESCRIPTION
**Pull request by mrednic-1A**
## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #782

### CODE REVIEW

**Review by matthieu-crouzet — DISMISSED** [review]

File: packages/@o3r/core/builders/run-script/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    execSync(`${pmr} ${options.script}`, {
+      stdio: ['inherit', 'inherit', 'inherit'],
+      cwd: projectRoot,
+      env: {
+        ...process.env
+      }
+    });
+    return { success: true };
+  }
+  catch (e) {
```
[the remark this answers is not in the document]
Comment (mrednic-1A) [reply packages/@o3r/core/builders/run-script/index.ts]: done

File: packages/@o3r/core/builders.json [hunk 2]
```diff
@@ -25,6 +25,11 @@
       "implementation": "./builders/ngc/",
       "schema": "./builders/ngc/schema.json",
       "description": "Perform a ngc build"
+    },
+    "run-script": {
```
[the remark this answers is not in the document]
Comment (mrednic-1A) [reply packages/@o3r/core/builders.json:29]: done

File: packages/@o3r/core/builders/run-script/index.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import * as fs from 'node:fs';
+import type { RunScriptBuilderSchema } from './schema';
+
+export default createBuilder<RunScriptBuilderSchema>(async (options, context): Promise<BuilderOutput> => {
+  context.reportRunning();
+  context.reportProgress(1, 1, `Running ${options.script} !!!`);
+  const specifiedRoot = context.target?.project && (await context.getProjectMetadata(context.target.project)).root?.toString();
+  const projectRoot = specifiedRoot ? resolve(context.workspaceRoot, specifiedRoot) : context.currentDirectory;
+  const angularJsonPath = join(context.workspaceRoot, 'angular.json');
+  const angularJson = JSON.parse(fs.readFileSync(angularJsonPath).toString());
```
Comment (kpanot) [inline packages/@o3r/core/builders/run-script/index.ts:15]: It is not something you can get from context ?

Comment (mrednic-1A) [reply packages/@o3r/core/builders/run-script/index.ts:15]: I could not find anything in context which exports it :(

Comment (kpanot) [reply packages/@o3r/core/builders/run-script/index.ts:15]: :'(
In that case you should check the if file exist and warn if it does not exist

**Review by matthieu-crouzet — DISMISSED** [review]

File: packages/@o3r/core/builders/run-script/index.ts (hunk 3 above)
Comment (mrednic-1A) [reply packages/@o3r/core/builders/run-script/index.ts]: Indeed. I have done the update.

File: packages/@o3r/core/builders/run-script/index.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import type { RunScriptBuilderSchema } from './schema';
+
+export default createBuilder<RunScriptBuilderSchema>(async (options, context): Promise<BuilderOutput> => {
+  context.reportRunning();
+  context.reportProgress(1, 1, `Running ${options.script} !!!`);
+  const specifiedRoot = context.target?.project && (await context.getProjectMetadata(context.target.project)).root?.toString();
+  const projectRoot = specifiedRoot ? resolve(context.workspaceRoot, specifiedRoot) : context.currentDirectory;
+
+  const angularJsonPath = join(context.workspaceRoot, 'angular.json');
+  let angularJson;
```
Comment (kpanot) [inline packages/@o3r/core/builders/run-script/index.ts:16]: not sure why the linter did not complain regarding the implicit `any`.
Can be written:
```typescript
const angularJson = fs.existsSync(angularJsonPath) ? JSON.parse(fs.readFileSync(angularJsonPath, {encoding: 'utf8'}) : undefined;
if (!angularJson) {
  context.logger.warn(`angular.json file cannot be found by @o3r/core:${context.builder.builderName} builder.
Detection of package manager runner will fallback on the one used to execute the actual command.`);
}
```

**Review by kpanot — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

File: packages/@o3r/core/builders/run-script/index.ts (hunk 4 above)
Comment (mrednic-1A) [reply packages/@o3r/core/builders/run-script/index.ts]: updated. I guess we have no error because `noImplicitAny` compiler option is true by default.

**Review by matthieu-crouzet — APPROVED** [review]

**Pull request merged**